### DISCUSSION TITLE
Range request support.

### DISCUSSION THREAD

**Discussion by Enferium** [nginxinc/nginx-s3-gateway#97] [Q&A]
Hello,

First of all, thank you for this great work. I was able to understand a lot of things about nginx and njs thanks to it.

It all works perfectly, but in my use case I need to make range requests on files of several dozen gigabytes. The standard behaviour of caching the whole file before sending the corresponding range request is therefore not desirable.

I was able to see the [link](https://www.nginx.com/blog/smart-efficient-byte-range-caching-nginx/) to this doc that you indicate in the default Template and the slice-by-slice caching method would be perfect.

I tried to add this feature unfortunately without success. The behaviour of nginx remains the same: download the whole file before sending the requested part.

My adding to default.conf looks like this:

<pre>
    proxy_cache s3_cache;

    <b>slice 1m;</b>
    proxy_cache_valid 200 <b>206</b> 302 ${PROXY_CACHE_VALID_OK};
    proxy_cache_valid 404 ${PROXY_CACHE_VALID_NOTFOUND};
    proxy_cache_valid 403 ${PROXY_CACHE_VALID_FORBIDDEN};
    proxy_cache_methods GET HEAD;
    proxy_cache_convert_head off;
    proxy_cache_revalidate on;
    proxy_cache_background_update on;
    proxy_cache_lock on;
    proxy_cache_use_stale error timeout http_500 http_502 http_503 http_504;
    proxy_cache_key "$request_method$host$uri<b>$slice_range</b>";`
    <b>proxy_set_header Range $slice_range;
    proxy_http_version 1.1;</b>
</pre>

Am I missing any particular information to enable slice-by-slice caching, for example:
1) Can slice-by-slice caching work with the default mechanism offered in case there is no range request.

2) Do I need to add information to the docker build process.

3) Is my .conf configuration correct, do I need to add information in this default.conf or elsewhere.

**Comment by dekobon** [comment]
Hi @Enferium I'm surprised that this isn't working. There must be something fiddly that is keeping it from working. It will take a debugging effort to discover the root cause.
